### ISSUE TITLE
[Bug]: Cannot use \t (tab) in snippet when expandtab is set to true

### ISSUE THREAD

**Issue by Bekaboo** [L3MON4D3/LuaSnip#919]
Consider the following snippet:

```lua
s({ trig = 'test' }, t('\t'))
```

ideally this should insert a tab after expansion despite the value of `&expandtab`, however it will only insert a tab when `expandtab` is set to false and will insert spaces instead when `expandtab` is set to true.

**Comment by Bekaboo** [comment]
@L3MON4D3 I can see it is a nice feature, but is there an option to toggle it off?

My use case is a little weird --  I have an autocmd to toggle `expandtab` on when my cursor is behind the text and off when there is no text or only white spaces before my cursor, so I can always use tab for indentation and spaces for alignment.

Now please consider the following multi-line snippet:

```lua
    s({ trig = 'if' }, {
      t('if '),
      i(1, 'condition'),
      t({ ' then', '' }),
      t('\t'),
      i(2),
      t({ '', 'end' }),
    }),
```

when my cursor is behind the `if` statement:

```lua
  if|
```

the `expandtab` is on (temporarily), but I would expect the text node `t('\t')` to insert a literal tab after expansion, like this:

```lua
  if [condition] then
  \t    
  end
```

The real case is more complex where I use a function node to determine how many spaces/tabs should be inserted for indentation for edge cases (mostly for mixed tabs and spaces) such as `ts=8` while `sw=4`.

Thanks for your time and consideration!

**Comment by L3MON4D3** [comment]
Whew, that is quite the setup😅
Are you sure that toggling expandtab is the best way of going about this? Have you considered checking the conditions when tab is pressed, and then inserting the appropriate characters?
Regardless of that, Iirc we don't expand tabs in indentation, so you could work with `indentSnippetNode` I think. Maybe that even simplifies the functionNode-stuff you have currently(?)

**Comment by Bekaboo** [comment]
@L3MON4D3 Thanks for your suggestion! Currently I just restore the original `expandtab` option in the function node as a workaround and I works pretty good (although this approach is hacky and not ideal).

> Mhmm, I guess we can add some options for this though... If we add them to snip_expand, this can even be changed at runtime, which is kind of cool I guess.

Maybe we can add a global option in the setup function to allow user to opt-out this feature while keeping the possibility of override it in individual snippets? That will be cool and looking forward for the implementation.

**Comment by Bekaboo** [comment]
Yeah it is also great to have this option accessible only in the setup function, for the sake of simplicity.

**Comment by Bekaboo** [comment]
@L3MON4D3 Just to let you know.. Finally I found a solution to set `expandtab` shortly before hitting `<Tab>` in insert mode and restore the original `expandtab` just after the text has changed, so the auto-expandtab feature is no longer a problem for me. However, the idea of providing an option for users to opt-out the default auto-expandtab feature might still be useful (but I don't care that much about it right now), what do you think about it?

Love your work on luasnip.

**Comment by L3MON4D3** [comment]
I'll definitely still add it, I think there are some cases where it's very hard to work around (for example if a dynamicNode which actually depends on nodes inserts a tab)

> Love your work on luasnip.

Thank you :)

**Comment by KevinNitroG** [comment]
Hi, I'm working on go and using `noexpandtab`, is there a workaround for this?
